hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxySupport.java

```diff
@@ -262,7 +262,11 @@ public MapConfig getMapConfig() {
     @Override
     public void initialize() {
         initializeListeners();
-        initializeIndexes();
+        if (getNodeEngine().isStartCompleted()) {
+            initializeIndexes();
+        } else {
+            initializeLocalIndexes();
```

Comment:
Why this fixes HZ-1192?

Reply:
The cluster-wide index addition clashes with operation execution restrictions during recovery.

I think it is anyway wrong to perform index initialization cluster-wide anyway in `MapProxySupport#initialize` and I would remove the `initializeIndexes` call altogether. We should only concern ourselves with locally owned partitions in proxy initialization.
@ahmetmircik wdyt?

Reply:
Not sure what the reason was to create indexes before start-completed. 
Isn't it an option to throw exception if start is not completed yet?

Reply:
Had a look at this again, making this local only seems like a behavior change. With this change, instead of relying on operation system guarantees, remote nodes proxies will be created by eventing system guarantees. This can introduce unexpected changes in effective behavior, when eventing system is busy and it drops events.

Reply:
As discussed with Ahmet, I will prepare a separate PR for the [HZ-1192 fix](https://github.com/hazelcast/hazelcast/pull/21844/commits/f1e60e9f0c4346e0592b931799733243bc4e67c6), so it is easier to track and revert this commit before we merge this PR. For now, I am leaving the commit in to facilitate testing.

Reply:
extracted in #22485 -- I still keep the commit as part of this PR as there is still some testing ongoing with those branches. Will revert it before merge.